desk/lib/contacts.hoon

```diff
@@ -227,6 +230,21 @@
       ==
     |
   &
+::  +sane-nickname: validate a nickname against network id
+::
+++  sane-nickname
+  |=  [src=(unit @p) txt=@t]
+  ^-  ?
+  ::  extract initial codepoint
+  =+  c=(end [3 (teff txt)] txt)
+  ?.  (~(has in con-sig:confusable:unicode) c)  &
+  ::  disallow spoofing sig
+  ?.  =('~' c)  |
+  ::  ~nickname, enforce urbit id
+  ::TODO allow extended nicknames, eg. '~zod🤴'?
+  ?~  ship=(rush txt ;~(pfix sig fed:ag))  |
+  ?~  src  &
+  =(u.src u.ship)
```

Comment:
Your initial intuition here (about wanting to run `(norm-p txt)` first) was correct, and I think other suggestions/this implementation is probably too strict.

For the `TODO`, wrap the ship name parsing in `;~(sfix ... (star next))`